src/Data/List/Relation/Unary/Grouped.agda

```diff
@@ -0,0 +1,41 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Property that elements are grouped
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.List.Relation.Unary.Grouped where
+
+open import Data.List using (List; []; _∷_; map)
+open import Data.List.Relation.Unary.All as All using (All; []; _∷_; all)
+open import Data.Sum using (_⊎_; inj₁; inj₂)
+open import Data.Product using (_×_; _,_)
+open import Relation.Binary as B using (REL; Rel)
+open import Relation.Unary as U using (Pred)
+open import Relation.Nullary using (¬_; yes)
+open import Relation.Nullary.Decidable as Dec
+open import Relation.Nullary.Sum using (_⊎-dec_)
+open import Relation.Nullary.Product using (_×-dec_)
+open import Relation.Nullary.Negation using (¬?)
+open import Level using (_⊔_)
+
+infixr 5 _∷_ _∷≈_
+
+data Grouped {a ℓ} {A : Set a} (_≈_ : Rel A ℓ) : Pred (List A) (a ⊔ ℓ) where
+  [] : Grouped _≈_ []
+  _∷_ : ∀ {x xs} → All (λ y → ¬ (x ≈ y)) xs → Grouped _≈_ xs → Grouped _≈_ (x ∷ xs)
+  _∷≈_ : ∀ {x y xs} → x ≈ y → Grouped _≈_ (y ∷ xs) → Grouped _≈_ (x ∷ y ∷ xs)
+
+module _ {a ℓ} {A : Set a} {_≈_ : Rel A ℓ} where
+  grouped : B.Decidable _≈_ → U.Decidable (Grouped _≈_)
```

Comment:
(`All` and `Any` notwithstanding, and which need to be made consistent with rest), the other predicates and relations call this property `grouped?` rather than `grouped`.